### PR TITLE
#257. Fix chunk begins of split fields

### PR DESCRIPTION
**Pull request by avbelov23**

### CODE REVIEW

File: tempesta_fw/http_parser.c [hunk 1]
```diff
@@ -101,9 +101,16 @@ st: __attribute__((unused)) 						\
 	TFW_DBG("parser: " #st "(%d:%d): c=%#x(%c), r=%d\n",		\
 		st, parser->_i_st, c, isprint(c) ? c : '.', r);
 
-#define __FSM_EXIT(field)						\
+#define __FSM_EXIT(p)							\
```
Comment (krizhanovsky) [inline tempesta_fw/http_parser.c]: ```p``` is used as local variable in all FSM functions, so the change isn't good.

File: tempesta_fw/http_parser.c [hunk 2]
```diff
@@ -101,9 +101,16 @@ st: __attribute__((unused)) 						\
...
 do {									\
-	if (field) { /* staticaly resolved */				\
+	if (p) { /* staticaly resolved */				\
+		TfwStr *field = (p);					\
```
Comment (krizhanovsky) [inline tempesta_fw/http_parser.c]: Why do we need the temporary variable ```field```?

Comment (avbelov23) [inline tempesta_fw/http_parser.c]: Sametimes as field we pass NULL (void*) and we can not do field->ptr.

Comment (avbelov23) [inline tempesta_fw/http_parser.c]: Sometimes as ''field we pass NULL(void) and we can't do ''field->ptr

**Comment by krizhanovsky** [comment]
As discussed ____FSM_MOVE_LAMBDA() calls tfw_http_msg_hdr_chunk_fixup() for uninitialized parser->hdr even for URI and Host parsing, which is surely wrong. The reason for the bug is that tfw_http_msg_set_data() (usually called from tfw_http_msg_hdr_chunk_fixup()) is not called for parser->uri_path. So the merge request is wrong and higher level logic must be fixed.

**Pull request closed without merging**